plugins/callback/foreman.py

```diff
@@ -106,30 +161,29 @@
 from ansible.plugins.callback import CallbackBase
 
 
-def build_log(data):
+def build_log_legacy(data):
     """
     Transform the internal log structure to one accepted by Foreman's
     config_report API.
     """
-    for source, msg in data:
-        if msg.get('failed'):
-            level = 'err'
-        elif msg.get('changed'):
-            level = 'notice'
-        else:
-            level = 'info'
-
-        yield {
-            "log": {
-                'sources': {
-                    'source': source,
-                },
-                'messages': {
-                    'message': json.dumps(msg, sort_keys=True),
-                },
-                'level': level,
-            }
+    if data.get('failed'):
```

Comment:
Not sure what you mean, items is a list but it contains dicts now instead of lists. Previously it was a list of lists, now it is a list of dicts. Is there an error or exception you see?

Reply:
I still see this error:

```
[WARNING]: Failure using method (v2_playbook_on_stats) in callback plugin (<ansible.plugins.callback.foreman.CallbackModule object at 0x7fe990b01810>): 'list' object has no attribute 'get'
Callback Exception: 
  File "/usr/lib/python2.7/site-packages/ansible/executor/task_queue_manager.py", line 381, in send_callback
    method(*new_args, **kwargs)
   File "/home/vagrant/foreman-ansible-modules/plugins/callback/foreman.py", line 386, in v2_playbook_on_stats
    self.send_reports(stats)
   File "/home/vagrant/foreman-ansible-modules/plugins/callback/foreman.py", line 341, in send_reports
    self.send_reports_foreman_legacy(stats)
   File "/home/vagrant/foreman-ansible-modules/plugins/callback/foreman.py", line 327, in send_reports_foreman_legacy
    "logs": list(build_log_legacy(self.items[host])),
   File "/home/vagrant/foreman-ansible-modules/plugins/callback/foreman.py", line 133, in build_log_legacy
    if data.get('failed'):
```

Reply:
When do you see this error? When in foreman (legacy) mode? It works for me.

Reply:
Yeap, still the same error. Used `report_type = foreman` and used both custom playbook and `-m ansible.builtin.copy -a "src=/etc/hosts dest=/tmp/hosts"`, same results.

Reply:
Ok now I am getting it. Will fix.